dio/test/http_client_adapter_test.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:dio/dio.dart';
+import 'package:dio/io.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group(HttpClientAdapter, () {
+    test(
+        'IOHttpClientAdapter.onHttpClientCreate is only executed once per request',
+        () async {
+      int onHttpClientCreateInvokeCount = 0;
+      final dio = Dio();
+      dio.httpClientAdapter = IOHttpClientAdapter(onHttpClientCreate: (client) {
+        onHttpClientCreateInvokeCount++;
+        return client;
+      });
+      await dio.get('https://pub.dev');
+      expect(onHttpClientCreateInvokeCount <= 1, isTrue);
```

Comment:
This should be exactly 1?
```suggestion
      expect(onHttpClientCreateInvokeCount, 1);
```

Reply:
This might be a semantic issue, the unit test is 'IOHttpClientAdapter.onHttpClientCreate is only executed once per request', I think it would be better to use '<=1' here. Of course, using '1' is also fine.